### ISSUE TITLE
unable to remove azurerm_sql_database from elastic pool

### ISSUE THREAD

**Issue by paulmchenry** [hashicorp/terraform-provider-azurerm#2369]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
Terraform version: 0.11.10
provider.azurerm v1.19.0

### Affected Resource(s)

* azurerm_sql_database

### Terraform Configuration Files

```hcl
resource "azurerm_sql_elasticpool" "dev-we-sqltest-ep" {
  name                = "dev-we-sqltest-ep"
  resource_group_name = "${azurerm_resource_group.dev-we-sql.name}"
  location            = "${azurerm_resource_group.dev-we-sql.location}"
  server_name         = "${azurerm_sql_server.dev-we-sqltest.name}"
  edition             = "Standard"
  dtu                 = 100
}
resource "azurerm_sql_database" "rdtest-elastic-pool-db2" {
  name                = "rdtest-elastic-pool-db2"
  resource_group_name = "${azurerm_resource_group.dev-we-sql.name}"
  location            = "${azurerm_resource_group.dev-we-sql.location}"
  server_name         = "${azurerm_sql_server.dev-we-sqltest.name}"
  elastic_pool_name                = "${azurerm_sql_elasticpool.dev-we-sqltest-ep.name}"
  edition                          = "Standard"
  requested_service_objective_name = "ElasticPool"
  collation                        = "Latin1_General_CI_AS"
}
```

### Expected Behavior
Terraform recognises when the elastic pool has been commented out and moves the database out of the pool.

### Actual Behavior
Nothing - Terraform shows no changes to be applied when trying to remove a database from an elastic pool. 

### Steps to Reproduce
1. Create a database that is part of an elastic pool.
2. Remove/comment out `elastic_pool_name`.
3. Run `terraform apply`

### Important Factoids
I first noticed this error when trying to move the database out of the elastic pool and switching the `requested_service_objective_name` to `S1`, from `ElasticPool`, which returned the following error:

`sql_database.rdtest-elastic-pool-db2: sql.DatabasesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: json: cannot unmarshal array into Go struct field serviceError2.details of type map[string]interface {}`

After looking in the Azure portal I found that this was the error being returned:

`Elastic pool 'dev-we-sqltest-ep' and service level objective 'S1' combination is invalid.`

This caused me to realise that the elastic pool change was either not being made, or being made in the wrong order, after which I tested just trying to remove the database from the pool without changing the service tier parameter, which confirmed that no changes were being attempted to the elastic pool.

### References
Previously a similar issue was raised [here](https://github.com/terraform-providers/terraform-provider-azurerm/issues/590) though the referenced issues within this ticket appear to be more about missing values from the `requested_service_objective_name` and problems relating to the `requested_service_objective_id`, so it seems to be a different problem from this one. I've just linked to it as at first glance it looks like the same issue and that it may be resolved.

**Comment by RobotEdward** [comment]
I encountered this issue today as well and was able to resolve with an odd workaround.

**Scenario:**

1. Database exists and is in an elastic pool. 
2. I wanted to move it to a be Standard S0 instance by setting:
```
  elastic_pool_name                = ""
  edition                          = "Standard"
  requested_service_objective_name = "S0"
```
3. Error on apply: "Elastic pool 'DefaultPool' and service level objective 'S0' combination is invalid."

The same problem happened if I set elastic_pool_name to null. 
It looks like both are being interpreted by Terraform or the AzureRM Provider as being 'not set' and hence leaving the old elastic pool name in place.

**Workaround**
Setting the elastic_pool_name to a single space seemed to have the desired effect of removing the existing pool name and my changes where applied successfully.
```
  elastic_pool_name                = " "
  edition                          = "Standard"
  requested_service_objective_name = "S0"
```
